x/bank/types/send_authorization.go

```diff
@@ -45,5 +77,26 @@ func (a SendAuthorization) ValidateBasic() error {
 	if !a.SpendLimit.IsAllPositive() {
 		return sdkerrors.ErrInvalidCoins.Wrapf("spend limit must be positive")
 	}
+
+	found := make(map[string]bool, 0)
+	for i := 0; i < len(a.AllowList); i++ {
+		if found[a.AllowList[i]] {
+			return ErrDuplicateEntry
+		} else {
+			found[a.AllowList[i]] = true
+		}
+	}
 	return nil
 }
+
+func validateAllowedAddresses(allowed []sdk.AccAddress) ([]string, error) {
```

Comment:
this naming is bit confusing. We are not validating anything inside. How about `toBech32Addresses()`